### PR TITLE
[LLVMCPU] Add a flag to disable distribution.

### PR DESCRIPTION
**Pull request by hanhanW**

### CODE REVIEW

**Review by MaheshRavishankar — CHANGES_REQUESTED** [review]
I am not sure its great to add such an option.... This is for experimentation, and transform dialect path seems like the best way. This creates too many branch points that will be hard to maintain.

**Comment by dcaballe** [comment]
Thanks for uploading this so quickly, Hanhan! 

Mahesh, I was discussing with Hanhan that I tried the code below and seemed to be enough. I think having this kind of flag would be useful in general, even we mark is as a debug/experimental flag.

```
@@ -358,7 +364,14 @@ static SmallVector<int64_t> getDefaultDistributedLoopTileSizes(
   assert(lbs.size() == ubs.size() && lbs.size() == minTileSizes.size() &&
          lbs.size() == maxTileSizes.size() &&
          "expected all vectors to be of equal size");
+
   size_t numDims = lbs.size();
+  // Set all the distribution tile sizes to zero if thread distribution is
+  // disabled.
+  if (!clEnableThreadDistribution) {
+    return SmallVector<int64_t>(numDims, 0);
+  }
+
   SmallVector<int64_t> distributedTileSizes(numDims, 1);
   SmallVector<int64_t> numWorkgroupsPerDim(numDims, 1);
   SmallVector<int64_t> workload(numDims, 1);
```

**Review by MaheshRavishankar — APPROVED** [review]
Alright, Ill defer to you guys.

File: compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp [hunk 1]
```diff
@@ -50,9 +50,15 @@ static llvm::cl::opt<int> clNativeVectorSizeInBytes(
 
 static llvm::cl::opt<int> clNumberOfRuntimeThreads(
     "iree-codegen-llvm-number-of-threads",
-    llvm::cl::desc("number of threads that are used at runtime"),
+    llvm::cl::desc("number of threads that are used at runtime if codegen "
+                   "thread distribution is enabled"),
     llvm::cl::init(8));
 
+static llvm::cl::opt<bool> clEnableDistribution(
+    "iree-codegen-llvm-enable-distribution",
```
Comment (MaheshRavishankar) [inline compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp]: Nit: I'd rather make this `disable-distribution`. We do distribution by default. So if you want to disable it you need to do something.

**Review by dcaballe — APPROVED** [review]

**Pull request merged**